vizro-core/src/vizro/cards/_cards.py

```diff
@@ -0,0 +1,119 @@
+"""Module containing default card components."""
+
+from typing import Callable, Optional
+
+import dash_bootstrap_components as dbc
+import pandas as pd
+from dash import dcc, get_relative_path, html
+
+from vizro.models.types import capture
+
+
+# LQ: The data frame is not used in the function, but needs to be provided here. I guess that is unavoidable for now?
+@capture("card")
+def text_card(data_frame: pd.DataFrame, text: str) -> dbc.Card:
+    """Static text card."""
+    return dcc.Markdown(text, dangerously_allow_html=False)
+
+
+@capture("card")
+def nav_card(data_frame: pd.DataFrame, text: str, href: str) -> dbc.Card:
+    """Static navigation card."""
+    return dbc.NavLink(
+        dcc.Markdown(text, dangerously_allow_html=False),
+        href=get_relative_path(href) if href.startswith("/") else href,
+    )
+
+
+# Example 1: KPI Card with Markdown
+# (+) Allows for unlimited customisation on text
+# (-) Custom styling becomes difficult due to className provision
+@capture("card")
+def kpi_card_mkd(data_frame: pd.DataFrame, title: str, value: str, agg_fct: Callable = sum) -> dbc.Card:
+    """Dynamic text card in form of a KPI Card."""
+    # LQ: Think about exposing an argument that allows for custom formatting such as formatting as currency
+    value = round(agg_fct(data_frame[value]), 2)
+
+    return dcc.Markdown(
+        f"""
+        ## {title}
+
+        # {value}
+        """,
+        dangerously_allow_html=False,
+    )
+
+
+# Example 2: KPI Card with HTML
+# (-) Customisation on text requires a new captured Callable to be created
+# (+) Allows for indefinite custom styling
+
+
+# LQ: Do we want all of these arguments? `title` and `value` are required, but the rest is extra functionality that
+# we could also outsource to creating their own CapturedCallable with that logic. It would be just more cumbersome to create these.
+@capture("card")
+def kpi_card(
+    data_frame: pd.DataFrame,
+    title: str,
+    value: str,
+    icon: Optional[str] = None,
+    agg_fct: Callable = sum,
+    value_format: Optional[str] = None,
+) -> dbc.Card:
+    """Dynamic text card in form of a KPI Card."""
+    value = agg_fct(data_frame[value])
+
+    return [
+        html.Div(
+            [
+                html.P(icon, className="material-symbols-outlined") if icon else None,
+                html.H2(title),
+            ],
+        ),
+        html.P(value_format.format(value) if value_format else value),
+    ]
+
+
+# LQ: Not sure if the removal of classNames is a better approach. It seems more unstable as it depends
```

Comment:
Yeah, this makes sense I think. The problem with class names everywhere is it looks kind of complicated if someone does want to copy and paste the code to make their own KPI card function. But I guess if we do it all through tricky CSS selectors then they'd need to know how to modify those, which is more difficult for most users.

Overall maybe somewhere between how you had it before (class names everywhere) and now (minimal class names) is best but I leave it up to you what you think is best.

Reply:
@antonymilne—instead of introducing our own `className`s, I've now wrapped the relevant sub-components into pre-existing Bootstrap components such as `dbc.CardHeader`, `dbc.CardBody`, and `dbc.CardFooter`. Similar to the `dbc.Card`, it's essentially just returning an HTML.Div with a pre-selected className such as "card-body" and "card-header" that we then customise.

Essentially, we still provide `classNames` where we need them instead of selecting the elements via the component hierarchy, but at least all of these `classNames `should be apparent from the bootstrap components and are not coming from us.

It's